### PR TITLE
Cherry-pick changes for JIRA 2095

### PR DESCRIPTION
**Pull request by sbroeder**

### CODE REVIEW

**Comment by sureshsubbiah** [comment]
jenkins, retest

File: core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/RMInterface.java [hunk 1]
```diff
@@ -137,6 +163,130 @@ public RMInterface() throws IOException {
...
+          final HRegionLocation lv_location = location;
+          final HConnection lv_connection = connection;
+          compPool.submit(new RMCallable2(ts, lv_location, lv_connection ) {
+             public Integer call() throws IOException {
+                return pushRegionEpochX(ts, lv_location, lv_connection);
+             }
+          });
+          try {
+            result = compPool.take().get();
+          } catch(Exception ex) {
```
Comment (selvaganesang) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/RMInterface.java]: As part of Exception handling, there has been an effort to catch specific exceptions only. Can you please modify this code accordingly.

File: core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/RMInterface.java [hunk 2]
```diff
@@ -137,6 +163,130 @@ public RMInterface() throws IOException {
...
+                    return rpcCallback.get();
+                 }
+              };
+
+              Map<byte[], PushEpochResponse> result = null;
+              try {
+                 if (LOG.isTraceEnabled()) LOG.trace("pushRegionEpochX -- before coprocessorService: startKey: "
+                     + new String(startKey, "UTF-8") + " endKey: " + new String(endKey, "UTF-8"));
+                 result = table.coprocessorService(TrxRegionService.class, startKey, endKey, callable);
+              } catch (Throwable e) {
```
Comment (selvaganesang) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/RMInterface.java]: ditto

File: core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/RMInterface.java [hunk 3]
```diff
@@ -261,7 +419,13 @@ static public void clearTransactionStates(final long transactionID) {
     static public synchronized void unregisterTransaction(final long transactionID) {
       TransactionState ts = null;
       if (LOG.isTraceEnabled()) LOG.trace("Enter unregisterTransaction txid: " + transactionID);
+      try {
         ts = mapTransactionStates.remove(transactionID);
+      } catch (Exception e) {
```
Comment (selvaganesang) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/RMInterface.java]: ditto

**Comment by sbroeder** [comment]
Yes, I can certainly make these changes.  I have some other rework to do due to test failures. I will close this and re-open when changes are complete.

**Pull request closed without merging**